src/grt/grt-wave_options-parse.adb

```diff
@@ -0,0 +1,323 @@
+      else
+         Tree_Index := Pkg;
+      end if;
+      Tree_Cursor := Trees (Tree_Index);
+      Previous_Tree_Cursor := null;
+
+      loop
+         First := Last;
+
+         loop -- Find next identifier
+            if Line (Last) = Sep (Tree_Index) then
+               Last := Last - 1;
+               exit;
+            elsif Last = Line'Last then
+               exit;
+            end if;
+            Last := Last + 1;
+         end loop;
+
+         Check_Validity (Line (First .. Last));
+         Tree_Updated := Update_Tree (Line (First .. Last), Tree_Index);
+         Line_Context.Max_Level := Line_Context.Max_Level + 1;
+
+         if Last = Line'Last then
+            if not Tree_Updated then
+               Error_Context ("ignored already known signal path", Warning);
+            end if;
+            return;
+         end if;
+
+         Last := Last + 2; -- Skip the separator
+         if Last > Line'Last then -- catch signal paths ending with /
+            Error_Context ("invalid signal path");
+         end if;
+
+      end loop;
+
+   end Parse_Path;
+
+   function Update_Tree (Elem_Name : String; Tree_Index : Tree_Index_Type)
+            return Boolean is
+      Sibling_Cursor : Elem_Acc := Tree_Cursor;
+      Previous_Sibling_Cursor : Elem_Acc := null;
+      Elem : Elem_Acc;
+   begin
+      loop
+         -- Already reached the last sibling and current identifier corresponds
+         -- to no existing element ? Then we will create an element
+         if Sibling_Cursor = null then
+            Elem := new Elem_Type'(Name => new String'(Elem_Name),
+                                   Line_Context => Line_Context,
+                                   Kind => Not_Found,
+                                   Next_Sibling | Next_Child => null);
+            if Previous_Sibling_Cursor = null then -- First element of level ?
+               if Previous_Tree_Cursor = null then -- Is a top_level ?
+                  Trees (Tree_Index) := Elem;
+               else
+                  Previous_Tree_Cursor.Next_Child := Elem;
+               end if;
+            else
+               Previous_Sibling_Cursor.Next_Sibling := Elem;
+            end if;
+            Previous_Tree_Cursor := Elem;
+            Tree_Cursor := null; -- Point to Elem.Next_Child which is null
+            return True;
+         -- Identifier was found in the tree ? Then move to its first child
+         elsif Elem_Name = Sibling_Cursor.Name.all then
+            Previous_Tree_Cursor := Sibling_Cursor;
+            Tree_Cursor := Sibling_Cursor.Next_Child;
+            return False;
+         end if;
+         Previous_Sibling_Cursor := Sibling_Cursor;
+         Sibling_Cursor := Sibling_Cursor.Next_Sibling;
+      end loop;
+   end Update_Tree;
+
+   procedure Check_Validity (Elem_Name : String) is
+   begin
+      if Elem_Name'Length = 0 then
+         Error_Context ("invalid signal path");
+      end if;
+      for Index in Elem_Name'Range loop
```

Comment:
You are right, it's useless, so I fixed it.

I have nevertheless added some more checks for path validity that won't 
be reported otherwise.